src/domain/app/header/__tests__/HeaderNotification.test.tsx

```diff
@@ -104,23 +101,20 @@ it('saves notification state to local storage', async () => {
   const localStorageObject = JSON.parse(
     localStorage.getItem(NOTIFICATION_STORAGE_KEY) as string
   );
-  await act(() => wait(100));
+  await sleep(100);
 
   expect(localStorageObject).toEqual({
     isVisible: true,
     closedNotificationHash: null,
   });
 
-  await act(
-    async () =>
-      await userEvent.click(
-        screen.getByRole('button', {
-          name: 'Sulje huomiotiedote',
-        })
-      )
+  await userEvent.click(
+    screen.getByRole('button', {
+      name: 'Sulje huomiotiedote',
+    })
   );
 
-  await act(() => wait(100));
+  await sleep(100);
```

Comment:
Could waitfor be used instead of sleep? That would make the test wait for certain condition, instead of just sleep for a while. Using sleep might fail sometimes when execution takes more time than expected.